pkg/config/validation.go

```diff
@@ -401,6 +401,22 @@ func validateTLSOptions(tls *networking.Server_TLSOptions) (errs error) {
 		return
 	}
 
+	if tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		// ISTIO_MUTUAL TLS mode uses either SDS or default certificate mount paths
+		// therefore, we should fail validation if other TLS fields are set
+		if tls.ServerCertificate != "" {
+			errs = appendErrors(errs, fmt.Errorf("ISTIO_MUTUAL TLS cannot have associated server certificate"))
```

Comment:
are you sure this is valid? People from salesforce wanted customizability of the path to the mutual TLS certificates (I guess they can use MUTUAL mode instead of ISTIO_MUTUAL.). Thoughts @ramaraochavali / @drichelson ?

Reply:
Yes, `MUTUAL` is used for that case for customization. `ISTIO_MUTUAL` is used for istio managed and reducing the config you need to type.